gateway/src/resty/http/proxy.lua

```diff
@@ -94,15 +95,17 @@ local function connect_proxy(httpc, request, skip_https_connect)
     ngx.log(ngx.DEBUG, 'connection to ', proxy_uri.host, ':', proxy_uri.port, ' established',
         ', pool: ', options.pool, ' reused times: ', httpc:get_reused_times())
 
+    ngx.log(ngx.DEBUG, 'targeting server ', uri.host, ':', uri.port)
+
     if uri.scheme == 'http' then
         -- http proxy needs absolute URL as the request path
-        request.path = format('%s://%s:%s%s', uri.scheme, host, port, uri.path or '/')
+        request.path = format('%s://%s:%s%s', uri.scheme, uri.host, uri.port, uri.path or '/')
         return httpc
     elseif uri.scheme == 'https' and skip_https_connect then
-        request.path = format('%s://%s:%s%s', uri.scheme, host, port, request.path or '/')
-        return _connect_tls_direct(httpc, request, host, port)
+        request.path = format('%s://%s:%s%s', uri.scheme, uri.host, uri.port, request.path or '/')
```

Comment:
I think that this will work, but please test this with routing policy, I remember that we had issues with that. Also, with Camel policy

Reply:
I will check with the routing policy yep! However, the camel policy I might need to get some help with because I don't understand how that works and the tests for that specific policy are not passing even without my changes. It seems like something is broken there or maybe it's expected and we need to test in another way?